app/questionnaire/value_source_resolver.py

```diff
@@ -100,36 +100,58 @@ def _resolve_list_item_id_for_value_source(
             else None
         )
 
+    def _resolve_repeating_answers_for_list(
+        self, *, answer_id: str, list_name: str
+    ) -> list[AnswerValueTypes]:
+        """Return the list of answers in answer store that correspond to the given list name and dynamic/repeating answer_id"""
+        answer_values: list[AnswerValueTypes] = []
+        for list_item_id in self.list_store[list_name]:
+            if answer_value := self._get_answer_value(
```

Comment:
I see the existing code for dynamic was like this, but this wouldn't work for falsey values like `0`.

Reply:
Fixed and updated test 👍